processor/test/integration/test_projects.js

```diff
@@ -31,7 +31,7 @@ describe('projects API', () => {
   beforeEach(async () => {
     await exec(`mkdir -p ${tmpDir}`)
     await exec(`mkdir -p ${repoDir}`)
-    this.app = createApp(repoDir, processingManagerStub)
+    this.app = createApp(repoDir, isRepoReadyForProcessingStub)
     this.supertest = supertest(this.app)
   })
 
```

Comment:
That stub is really ugly, can you not let it be `undefined` and check it down the line?